### PR TITLE
#166605105 Fix-Removes ids from seeders file and fix bug with activating/deactivating a user

### PR DESCRIPTION
**Pull request by Lundii**
#### What does this PR do?
* It removes ids from user seeder file
* It fixes a bug when activating or deactivating a user
#### Description of Task to be completed?
N/A
#### How should this be manually tested?
* login as a super admin
* activate or deactivate a user
#### Any background context you want to provide?
* initially the user is either activated nor deactivated even though a response of 200 is sent back
#### What are the relevant pivotal tracker stories?
[#166605105](https://www.pivotaltracker.com/story/show/166605105)
#### Screenshots (if appropriate)
N/A
#### Questions:
No

### CODE REVIEW

**Review by Adekoreday — CHANGES_REQUESTED** [review]
remove auto incremented feature in role model

**Review by Adekoreday — DISMISSED** [review]
you also need to look into your user controller line 32 should be in  an if statement

**Comment by Lundii** [comment]
> you also need to look into your user controller line 32 should be in an if statement

The if statement is not necessary, the findUser function already handles if the user is not in the database.
What do you suggest should be in the if statement?

true the response will be returned , but the program still throws message cant set headers after it has been sent indicating the program still runs after the return in the if. Kindly look into this

**Comment by Lundii** [comment]
> remove auto incremented feature in role model

I didn't create it, Its being like that since it was created. Why do you think it should be removed?

**Comment by allebd** [comment]
Kindly update the **verified** status of a created user to true, when the user is created by the super admin

**Review by allebd — COMMENTED** [review]
Kindly update the **verified** status of a created user to true, when the user is created by a super admin in this PR

**Comment by Lundii** [comment]
> Kindly update the **verified** status of a created user to true, when the user is created by a super admin in this PR

I looked into it, instead of setting the status to be verified, I decided to send a verification link to the newly created users email, so they can verify their account from there.

**Review by Adekoreday — CHANGES_REQUESTED** [review]
you need to enable super admin to create authorName

**Review by Adekoreday — CHANGES_REQUESTED** [review]
the links in the messages are not well formatted they need to be on a single line so the whole url will be sent

File: server/helpers/messages.js [hunk 1]
```diff
@@ -5,21 +5,38 @@ const signupMessage = (firstName, token) => {
...
+};
+const createAdminMessage = (user, role) => {
+  const message = {
+    text: 'hello Books',
+    html: `  <div style="background-color:#7AA9EF; color:#fff; padding:20px"> <span style="text-align: left; margin-right: 10px">HB</span>
+             <span style= "text-align:center">Welcome to hello books ${user.firstName.toUpperCase()}</span></div>
+             <div style="margin-top:30px" >  You have been added as an ${role}. Your login details are email: ${user.email}
+             password: password@1.</div>
+             <div>
+                 <p>Login or register to change your password immediately
```
Comment (henryade) [inline server/helpers/messages.js]: Remove register

File: server/helpers/messages.js [hunk 2]
```diff
@@ -5,21 +5,38 @@ const signupMessage = (firstName, token) => {
...
     text: 'hello Books',
     html: `  <div style="background-color:#7AA9EF; color:#fff; padding:20px"> <span style="text-align: left; margin-right: 10px">HB</span>
              <span style= "text-align:center">Welcome to hello books ${user.firstName.toUpperCase()}</span></div>
              <div style="margin-top:30px" >  You have been added as a/an ${role}. Your login details are email: ${user.email}
-             password: password@1. Please login and change your password immediately</div>
+             password: password@1.</div>
+             <div> Click on this link to confirm your email <a href=https://hello-books-staging.herokuapp.com/api/v1/auth/verify?token=${token}>
+             LINK </a></div >
+             <div>
+                 <p>Login or register to change your password immediately
```
Comment (henryade) [inline server/helpers/messages.js]: Remove register

**Review by henryade — CHANGES_REQUESTED** [review]

**Review by Adekoreday — COMMENTED** [review]
in the signup message kindly read the message there are some typos

File: server/helpers/messages.js (hunk 1 above)
Comment (Lundii) [reply server/helpers/messages.js]: Done

File: server/helpers/messages.js (hunk 2 above)
Comment (Lundii) [reply server/helpers/messages.js]: Done

**Comment by Lundii** [comment]
> you need to enable super admin to create authorName

I already gave superadmin the permission to create authors

**Comment by Lundii** [comment]
> the links in the messages are not well formatted they need to be on a single line so the whole url will be sent

Done. Check it out

**Review by Adekoreday — CHANGES_REQUESTED** [review]
authorize admin in admin.js should not be declared kindly put in the middleware function 
admin.post(authorizeUser(['superadmin', 'admin']))

**Comment by Lundii** [comment]
> authorize admin in admin.js should not be declared kindly put in the middleware function
> admin.post(authorizeUser(['superadmin', 'admin']))

Okay. Done

**Review by Adekoreday — DISMISSED** [review]
welldone

**Review by Nennyfills — DISMISSED** [review]
Great work

File: server/helpers/messages.js [hunk 3]
```diff
@@ -2,24 +2,39 @@ const signupMessage = (firstName, token) => {
   const message = {
     text: 'hello Books',
```
Comment (henryade) [inline server/helpers/messages.js]: hello books should be capitalized

File: server/helpers/messages.js [hunk 4]
```diff
@@ -2,24 +2,39 @@ const signupMessage = (firstName, token) => {
...
+             <div>Click on this link to confirm your registration <a href=https://hello-books-staging.herokuapp.com/api/v1/auth/verify?token=${token}> LINK </a></div>
      `,
   };
   return message;
 };
 
-const createUserMessage = (user, role) => {
+const createUserMessage = (user, role, token) => {
   const message = {
     text: 'hello Books',
```
Comment (henryade) [inline server/helpers/messages.js]: here too

File: server/helpers/messages.js [hunk 5]
```diff
@@ -2,24 +2,39 @@ const signupMessage = (firstName, token) => {
...
     html: `  <div style="background-color:#7AA9EF; color:#fff; padding:20px"> <span style="text-align: left; margin-right: 10px">HB</span>
              <span style= "text-align:center">Welcome to hello books ${user.firstName.toUpperCase()}</span></div>
              <div style="margin-top:30px" >  You have been added as a/an ${role}. Your login details are email: ${user.email}
-             password: password@1. Please login and change your password immediately</div>
+             password: password@1.</div>
```
Comment (henryade) [inline server/helpers/messages.js]: i thought the password is either 'setpassword or an enviroment variable?

File: server/helpers/messages.js [hunk 6]
```diff
@@ -2,24 +2,39 @@ const signupMessage = (firstName, token) => {
...
+             <div>
+                 <p>Login to change your password immediately
+             </div>
+     `,
+  };
+  return message;
+};
+const createAdminMessage = (user, role) => {
+  const message = {
+    text: 'hello Books',
```
Comment (henryade) [inline server/helpers/messages.js]: here too

File: server/helpers/messages.js [hunk 7]
```diff
@@ -2,24 +2,39 @@ const signupMessage = (firstName, token) => {
...
+    html: `  <div style="background-color:#7AA9EF; color:#fff; padding:20px"> <span style="text-align: left; margin-right: 10px">HB</span>
+             <span style= "text-align:center">Welcome to hello books ${user.firstName.toUpperCase()}</span></div>
+             <div style="margin-top:30px" >  You have been added as an ${role}. Your login details are email: ${user.email}
+             password: password@1.</div>
```
Comment (henryade) [inline server/helpers/messages.js]: i thought the password is either 'setpassword or an enviroment variable?

File: server/helpers/setupNewUser.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import responseMessage from './responseMessage';
+import createUserRole from './createUserRole';
+import { createUserMessage, createAdminMessage } from './messages';
```
Comment (henryade) [inline server/helpers/setupNewUser.js]: this shouldn't be destructed here

File: server/helpers/setupNewUser.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+import sendMail from './sendMail';
+
+/**
+ * Sets up a newly created user
+ * @param {object} response - the response object
+ * @param {object} userData - the userId
+ * @param {string} role - the user role
+ * @param {string} token - user token
+ * @return {null} - no returned value
```
Comment (henryade) [inline server/helpers/setupNewUser.js]: returns json

File: server/routes/admin.js [hunk 10]
```diff
@@ -11,14 +11,14 @@ const { verifyToken, authorizeUser, AuthorValidator } = middlewares;
 const { addAuthor } = AuthorController;
 const { changeUserStatus } = AdminController;
 const { authorValidation } = AuthorValidator;
-const authorizeAdmin = authorizeUser(['admin']);
 
-admin.post(`${BASE_URL}/author`, verifyToken, authorizeAdmin, authorValidation(), addAuthor);
+admin.post(`${BASE_URL}/author`, verifyToken, authorizeUser(['superadmin', 'admin']), authorValidation(), addAuthor);
```
Comment (henryade) [inline server/routes/admin.js]: no comment here

**Review by henryade — CHANGES_REQUESTED** [review]

File: server/helpers/messages.js (hunk 3 above)
Comment (Lundii) [reply server/helpers/messages.js]: Fixed

File: server/helpers/messages.js (hunk 4 above)
Comment (Lundii) [reply server/helpers/messages.js]: Fixed

File: server/helpers/messages.js (hunk 5 above)
Comment (Lundii) [reply server/helpers/messages.js]: Fixed

File: server/helpers/messages.js (hunk 6 above)
Comment (Lundii) [reply server/helpers/messages.js]: Fixed

File: server/helpers/messages.js (hunk 7 above)
Comment (Lundii) [reply server/helpers/messages.js]: Fixed

File: server/helpers/setupNewUser.js (hunk 8 above)
Comment (Lundii) [reply server/helpers/setupNewUser.js]: Fixed

File: server/helpers/setupNewUser.js (hunk 9 above)
Comment (Lundii) [reply server/helpers/setupNewUser.js]: I returned a function instead

File: server/routes/admin.js (hunk 10 above)
Comment (Lundii) [reply server/routes/admin.js]: Fixed

**Review by henryade — APPROVED** [review]

**Review by KvNGCzA — APPROVED** [review]

**Pull request merged**